libraries/engage/core/hooks/useFormState.js

```diff
@@ -30,16 +31,18 @@ export function useFormState(initialState, complete, validationConstraints = {})
       return;
     }
     if (valid === true) {
-      complete(values);
-      setChanged(false);
-      setSubmitting(false);
+      (async () => {
+        await complete(values);
```

Comment:
This causes a React warning. 

![Bildschirmfoto 2020-03-25 um 17 42 50](https://user-images.githubusercontent.com/13087792/77562247-13039700-6ec0-11ea-91f9-8ea853f0b0d0.png)

It seems that when this code is executed, the component is already unmounted, and state changes are not possible anymore. So maybe you could just remove the next two lines, since in any situation there will be a new stage rendered after the request is done. ProgressBar and Button are not visible anymore, so they don't need to be updated.

As an alternative i found a funny solution, i never thought about. Check the "A Simple Solution" section of https://dev.to/pallymore/clean-up-async-requests-in-useeffect-hooks-90h It provides a solution with the effect cleanup function together with a local variable within the effect. So the effect could look like
```javascript
   useEffect(() => {
    if (!isSubmitting) {
      return;
    }

    let mounted = true;

    if (valid === true) {
      (async () => {
        await complete(values);

        if (mounted) {
          setSubmitting(false);
          setChanged(false);
        }
      })();
    }

    return () => {
      mounted = false;
    };
  }, [complete, isSubmitting, values, valid]);
```